cmd/argo/commands/executorplugin/io.go

```diff
@@ -0,0 +1,90 @@
+package executorplugin
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"text/template"
+
+	apiv1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/yaml"
+
+	"github.com/argoproj/argo-workflows/v3/pkg/plugins/spec"
+)
+
+func loadPluginManifest(pluginDir string) (*spec.Plugin, error) {
+	manifest, err := os.ReadFile(filepath.Join(pluginDir, "plugin.yaml"))
+	if err != nil {
+		return nil, err
+	}
+	p := &spec.Plugin{}
+	err = yaml.UnmarshalStrict(manifest, p)
+	if err != nil {
+		return nil, err
+	}
+	files, err := filepath.Glob(filepath.Join(pluginDir, "server.*"))
+	if err != nil {
+		return nil, err
+	}
+	if len(files) == 1 {
+		code, err := os.ReadFile(files[0])
+		if err != nil {
+			return nil, err
+		}
+		p.Spec.Sidecar.Container.Args = []string{string(code)}
+	} else if len(files) > 1 {
+		return nil, fmt.Errorf("plugin %s has more than one server.* file", p.Name)
+	}
+	return p, p.Validate()
```

Comment:
Do we need the else statement here? 

```suggestion

        if len(files) > 1 {
		return nil, fmt.Errorf("plugin %s has more than one server.* file", p.Name)
	}
	if len(files) == 1 {
		code, err := os.ReadFile(files[0])
		if err != nil {
			return nil, err
		}
		p.Spec.Sidecar.Container.Args = []string{string(code)}
	} 
	return p, p.Validate()
```